src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/mapper/EnvelopeMapper.java

```diff
@@ -47,25 +51,35 @@ public static Envelope toDbEnvelope(InputEnvelope envelope, String containerName
             envelope.caseNumber,
             envelope.previousServiceCaseReference,
             envelope.classification,
-            toDbScannableItems(envelope.scannableItems),
+            toDbScannableItems(envelope.scannableItems, ocrValidationWarnings),
             toDbPayments(envelope.payments),
             toDbNonScannableItems(envelope.nonScannableItems),
             containerName
         );
     }
 
-    private static List<ScannableItem> toDbScannableItems(List<InputScannableItem> scannableItems) {
+    private static List<ScannableItem> toDbScannableItems(
+        List<InputScannableItem> scannableItems,
+        List<String> ocrValidationWarnings
+    ) {
         if (scannableItems != null) {
             return scannableItems
                 .stream()
-                .map(EnvelopeMapper::toDbScannableItem)
+                .map(item ->
+                    toDbScannableItem(
+                        item,
+                        item.ocrData != null ? ocrValidationWarnings.toArray(new String[0]) : null
```

Comment:
this will copy the same warnings to multiple scannable items there are more than 1 containing OCR data

Reply:
solved